ckan/tests/views/test_cache.py

```diff
@@ -0,0 +1,193 @@
+import pytest
+import re
+import hashlib
+from flask import Request, Response
+from werkzeug.test import EnvironBuilder
+
+from ckan.tests.helpers import CKANTestApp
+from werkzeug.test import TestResponse
+
+import ckan.views as views
+
+field_name = '_csrf_token'  # emulate WTF_CSRF_FIELD_NAME for regex verification check
+pattern = fr'(?i)((?:_csrf_token|{field_name})[^>]*?\b(?:content|value)=|\bnonce=)["\'][^"\']+(["\'])'  # noqa: E501
+
+
+def clean_dynamic_values(text):
+    return re.sub(pattern, lambda m: m.group(1) + '="etag_removed"', text)
+
+
+@pytest.mark.ckan_config("ckan.cache_expires", 3600)
+def test_sets_cache_control_headers_default(app: CKANTestApp):
+    """Test that cache control headers are set correctly when caching is allowed."""
+
+    builder = EnvironBuilder(path='/', method='GET', headers={}, environ_overrides={})
+    env = builder.get_environ()
+    Request(env)
+    response = Response()  # dummy response
+
+    with app.flask_app.request_context(env):  # only works if you have app.flask_app
+        updated_response = views.set_cache_control_headers_for_response(response)
+    assert updated_response.cache_control.public is True, updated_response
+    assert updated_response.cache_control.max_age == 3600, updated_response
+    assert updated_response.cache_control.must_revalidate is True, updated_response
+
+
+@pytest.mark.ckan_config("ckan.cache_expires", 3600)
+def test_sets_cache_control_headers_with__no_private_cache__set(app: CKANTestApp):
+    """Test that cache control headers are set correctly when caching is allowed."""
```

Comment:
This comment is exactly the same as the previous test. It should say something more specific to this test.